chaincode_example02 using C

kleash opened the pull request:
- init, query, invoke and delete function
- custom malloc as clang doesn't support system calls
- command to compile C to wasm using clang in readme

Resolves #17

kleash committed:
chaincode_example02 using C
- init, query, invoke and delete function
- custom malloc as clang doesn't support system calls
- command to compile C to wasm using clang in readme

Signed-off-by: Shubham Aggarwal <shubham6300@gmail.com>

MHBauer commented on sample-wasm-chaincode/README.md:
````diff
@@ -33,3 +33,22 @@ We have a sample wasm binary under: ```sample-wasm-chaincode/chaincode_example02
+## C
+We have a sample wasm binary under: ```sample-wasm-chaincode/chaincode_example02/c/app_main.wasm```
+
+ - Install clang8 and llvm [Linux repository link](https://apt.llvm.org)
````
clang8? extra 8?

kleash replied:
yeah, need to run it with clang v8

MHBauer commented on sample-wasm-chaincode/chaincode_example02/c/main.c:
```diff
@@ -0,0 +1,313 @@
+    }
+
+    return number;
+}*/
```
Do we need the commented out stuff?

kleash replied:
That's another way for converting char pointer to int. I was thinking to keep it and if we ran into some problem with current method, I will switch to this one..

swetharepakula replied:
If we don't need this, I am inclined to say delete it. Though I understand you wanting to be able to keep the code around. Is it possible to keep the commented code as a branch on your fork, so you have it in case we have issues with the implementation that is not commented out?

kleash replied:
Removed it. I copied it to another branch

MHBauer reviewed (commented):
Can we hook this in to the existing tests?

kleash commented:
> Can we hook this in to the existing tests?

Will do. I have one change stashed in local with renaming invoke to execute. Let me create that PR first and then I will add this.

Shall I add it in same wasmcc unit test(that test runs rust wasm) or is it better to make it as part of integration test?

MHBauer commented:
Depends on what it needs? I expect it to test similar to the rust test. Which I think I now realize is simply loading.

Both rust and C should be part of the integration test.

For this PR, a unit test should suffice. The integration tests should be a separate change.

kleash committed:
unit test case for C wasm example
- test wether C chaincode compiles to wasm
- install clang, llvm in go actions

Signed-off-by: Shubham Aggarwal <shubham6300@gmail.com>

kleash commented:
Added unit test case

kleash commented:
> Added unit test case

uh-oh!! need to install clang in github workflow

swetharepakula commented on Makefile:
```diff
@@ -20,5 +20,9 @@ bin/wasm-pusher:
 	mkdir -p bin/
 	cd tools/wasm-pusher && go build -o ../../bin/wasm-pusher
 
+bin/c_chaincode_example02:
```
maybe bin/c_chaincode ? Just thinking of picking something simpler that people could easily type. Or even making another target to this that is just c_chaincode?

kleash replied:
renamed to c_chaincode

swetharepakula reviewed (changes requested):
I think we should clean up the commented code if it is not being used in the example.  Otherwise code looks good to me. Only other suggestions is to rename the new make target to make it more user friendly.
